### PR TITLE
fix: clear unreached commits when commits are pushed successfully

### LINKED ISSUE #603: push operation doesn't reset unreachedCommits
## Overview

type: bug

Push operation doesn't reset unreachedCommits. It should reset unreachedCommits when success like repush operation.

### PR DESCRIPTION
**Pull request by imoans**
fix #603 

## Overview
Clear unreached commits when commits are pushed successfully in `middlewareHandler.commitAndPush` and `middlewareHandler.pushAndCommit`

In `middlewareHandler.push`, not clear unreached commits because commits are pushed by the number specified in the action payload, and all unreached commits may not be pushed.

see e5d67ff for fix
see 46499be for test

### CODE REVIEW

File: modules/redux/package.json [hunk 1]
```diff
@@ -20,7 +20,9 @@
   "scripts": {
     "build": "upbin tsc -p tsconfig.build.json",
     "clean": "rm -rf lib",
-    "test": "upbin nyc upbin mocha --require ts-node/register 'test/**/*.test.ts' --color always",
+    "test": "npm run test:spec && npm run test:integration",
+    "test:spec": "upbin nyc upbin mocha --require ts-node/register 'test/spec/**/*.test.ts' --color always",
+    "test:integration": "upbin nyc upbin mocha --require ts-node/register 'test/integration/**/*.test.ts' --color always",
```
Comment (1natsu172) [inline modules/redux/package.json:25]: 🆒 💯

File: modules/redux/test/spec/middleware.test.ts [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+import { MiddlewareHandler } from "../../src/middleware";
+
+const dbClient = createDbClient();
+const httpClient = createHttpClient(PORT);
+
+const { store, actions } = createPhenylRedux(httpClient);
+const sessionClient = createSessionClient(dbClient);
+const server = createServer(dbClient);
+
+describe("middlewareHandler", async () => {
```
Comment (1natsu172) [inline modules/redux/test/spec/middleware.test.ts:22]: middlewareテスト大感謝 🥇

**Review by 1natsu172 — APPROVED** [review]
LGTM

**Comment by imoans** [comment]
@1natsu172 
Thank you for your fix of test

**Review by sasurau4 — APPROVED** [review]
LGTM 👍

**Review by yamatatsu — APPROVED** [review]

**Pull request merged**